packages/meditrak-app/app/assessment/specificQuestions/EntityQuestion.js

```diff
@@ -43,6 +43,16 @@ const DumbEntityQuestion = props => {
     return <CodeGeneratorQuestion {...props} />;
   }
 
+  // RN-984 if selected entity is not in the list, clear the selected entity
+  const { selectedEntityId, validEntities } = props;
+  if (selectedEntityId) {
+    const [selectedEntity] = validEntities.filter(entity => entity.id === selectedEntityId);
```

Comment:
It seems like the 'find' method might do exactly what you're doing here, but without needing to destructure.
`const selectedEntity = validEntities.find(entity => entity.id === selectedEntityId);`

Reply:
Good spotting, updated